### ISSUE TITLE
LOG014 Is gives false positives

### ISSUE THREAD

**Issue by hmvp** [astral-sh/ruff#18726]
### Summary

LOG014 should report `exc_info=True` but also reports on `exc_info=(exc_type, exc_value, exc_traceback)` which is valid but because it is truthy it reports a lint error

### Version

ruff 0.12.0

**Comment by ntBre** [comment]
Thanks for the report! This looks like a departure from the upstream linter's behavior too:

```python
# try.py
import logging

logging.warning("warning", exc_info=(True,))
```

```shell
> uvx --with flake8-logging flake8 try.py --select LOG014
> echo $?
0
```

They also check if the value is an [`ast.Constant`](https://docs.python.org/3/library/ast.html#ast.Constant) ([source](https://github.com/adamchainz/flake8-logging/blob/dbe88e76947450a7c4c1ebfb7a7a44875004f410/src/flake8_logging/__init__.py#L334-L338)) whereas we only check if it's truthy:

https://github.com/astral-sh/ruff/blob/05e28451bec6f3a00750e1e1445f6f8b5cb8ebeb/crates/ruff_linter/src/rules/flake8_logging/rules/exc_info_outside_except_handler.rs#L101
